### ISSUE TITLE
Pass MPI hints to h5py.File? (h5py+mpi4py on lustre)

### ISSUE THREAD

**Issue by dajuno** [h5py/h5py#673]
Hello all, 

Is it possible to set mpi hints via mpi4py.MPI Info objects? 

Context: I want to use h5py in parallel with mpi4py on a LUSTRE system. When trying to create a h5py.File with driver='mpio' I get an error:
```
This requires fcntl(2) to be implemented. As of 8/25/2011 it is not. Generic MPICH Message: File locking failed in ADIOI_Set_lock(fd 3,cmd F_SETLKW/7,type F_RDLCK/0,whence 0) with return value FFFFFFFF and errno 26.
- If the file system is NFS, you need to use NFS version 3, ensure that the lockd daemon is running on all the machines, and mount the directory with the 'noac' option (no attribute caching).
- If the file system is LUSTRE, ensure that the directory is mounted with the 'flock' option.
ADIOI_Set_lock:: Function not implemented
ADIOI_Set_lock:offset 0, length 1
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```
Now, in some other but similar cases (hdf5 or netcdf + MPI on LUSTRE, like here: http://lists.mcs.anl.gov/pipermail/parallel-netcdf/2010-September/001106.html) where the trick was to set mpi hints:
```
MPI_Info_set(info, "romio_ds_write", "disable");
MPI_Info_set(info, "romio_ds_read",  "disable");
```

I guessed that in python this would translate to
```
mpi_info = MPI.Info.Create()                                                   
mpi_info.Set("romio_ds_write", "disable")                                      
mpi_info.Set("romio_ds_read", "disable") 
```
Can I pass the info to a h5py file? Naively I tried the obvious:
`f = h5py.File('parallel_test.hdf5', 'w', driver='mpio', comm=MPI.COMM_WORLD, info=mpi_info)`
but of course it doesn't work. 

Thanks for any advice..!

David

**Comment by andrewcollette** [comment]
Using `info` certainly should work... you might post to the mailing list (h5py at google groups) to ask if others have had similar problems.

**Comment by dajuno** [comment]
Ok, good to know! Thanks for the swift reply.

**Comment by andrewcollette** [comment]
Great... feel free to update or open a new issue if you do have problems with `info` after all.
